src/os/user/iterate_test_fgetent.go

```diff
@@ -0,0 +1,96 @@
+//go:build (aix || dragonfly || (!android && linux) || solaris) && cgo && !osusergo
+// +build aix dragonfly !android,linux solaris
+// +build cgo
+// +build !osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <errno.h>
+
+static void resetErrno(){
+	errno = 0;
+}
+
+static FILE* openUsersFile(){
+	FILE* fp;
+	fp = fopen("./testdata/users.txt", "r");
+	return fp;
+}
+
+static FILE* openGroupsFile(){
+	FILE* fp;
+	fp = fopen("./testdata/groups.txt", "r");
+	return fp;
+}
+*/
+import "C"
+
+import (
+	"os"
+)
+
+// iterateUsersHelperTest implements usersHelper interface and is used for testing
+// users iteration functionality with fgetpwent library call.
+type iterateUsersHelperTest struct {
+	fp *C.FILE
+}
+
+func (i *iterateUsersHelperTest) set() {
+	var fp *C.FILE
+	C.resetErrno()
+	fp, err := C.openUsersFile()
+	if err != nil {
+		panic(err)
+	}
+	i.fp = fp
+}
+
+func (i *iterateUsersHelperTest) get() (*C.struct_passwd, error) {
+	var result *C.struct_passwd
+	// fgetpwent returns ENOENT when there are no more records
```

Comment:
explain why the comment is there: this return value is undocumented in `fgetpwent(3)`. You may also redirect developers to docs of `fgetpwent_r(3)`, which glibc uses for `fgetpwent`.